docs/source/reference_guides/contact_models.rst

```diff
@@ -76,13 +76,34 @@ implement that school classes only meet on weekdays, make sick individuals stay
 
 A single variable or list of variables according to which the matching is assortative.
 All ``assort_by`` variables must be categorical. Individuals who have the same value in
-all ``assort_by`` variables belong to one group. The ``params`` DataFrame contains
-entries that govern the probability of meeting people from the own group. The index
-entry of that parameter values is
-``("assortative_matching", name_of_contact_model, variable_name)``.
-
-The remaining probability mass is spread on all other groups, adjusting for group sizes
-and number of planned contacts in each group.
+all ``assort_by`` variables belong to one group.
+
+Often the total number of groups (n_groups) is very high and the full group probability
+matrix, i.e. the matrix that describes how likely it is that an individual from group
+i meets someone from group j has n_groups * n_groups entries. Thus it is not possible
+to estimate this matrix precisely, without imposing some further structure.
+
+There are three ways of specifying the probabilities. In all cases, we first create one
+probability matrix per assort_by variable and combine them to the full group
+probability matrix, assuming independence.
+
+Here are the three ways of describing the per-variable matrices (starting with the most
+parsimonious one):
+
+1. Specifying one probability per assort_by variable. This is the probability of meeting
+   someone with the same value of the assort_by variable. The remaining probability mass
+   is spread uniformly over all other values. The params index of this probability is
+   ``("assortative_matching", name_of_contact_model, name_of_variable)``
+2. Specifying one probability per value of the assort_by variable. This is interpreted
+   as the diagonal of the per-variable probability matrix. The remaining probability
+   mass is spread uniformly on the off diagonal elements. In this case, the params index
+   is as follows: The first level is ``f"assortative_matching_{model_name}_{variable}"``.
+   The second and third level are the values of the variable, i.e. the second and third
+   level have to be identical.
+3. Specify the full per-variable probability matrix. In this case the first index level
+   is the same as in case 2. The second index level indicates the row label of the
+   probability matrix. The third index level indicates the column of the probability
+   matrix.
```

Comment:
Rows are "self" and columns are "other" if I see it right from the code. I would maybe write this here and explain that these matrices needn't be symmetrical (if group sizes vary).

Reply:
Done.